client/components/data/query-jetpack-modules/index.jsx

```diff
@@ -3,9 +3,12 @@ import { useEffect } from 'react';
 import { useDispatch } from 'react-redux';
 import { fetchModuleList } from 'calypso/state/jetpack/modules/actions';
 import isFetchingJetpackModules from 'calypso/state/selectors/is-fetching-jetpack-modules';
+import isSiteWpcomAtomic from 'calypso/state/selectors/is-site-wpcom-atomic';
 
 const request = ( siteId ) => ( dispatch, getState ) => {
-	if ( siteId && ! isFetchingJetpackModules( getState(), siteId ) ) {
+	const isAtomic = isSiteWpcomAtomic( getState(), siteId );
```

Comment:
What about non-atomic Jetpack sites? From what I recall, Jetpack sites (which aren't necessarily Atomic sites) also used the module data - extensively in site settings (but not only there).

Reply:
Good catch @tyxla! Adding support to Jetpack non-Atomic sites here: https://github.com/Automattic/wp-calypso/pull/91170/commits/2e34bb3a7e2ce8b3b0829a54ef4739f775a8f578